src/main/kotlin/org/rust/lang/core/resolve/ResolveEngine.kt

```diff
@@ -212,6 +209,26 @@ object ResolveEngine {
         return module.project.getPsiFor(module.cargoWorkspace?.findCrateByName(name)?.crateRoot)?.rustMod
     }
 
+    fun resolveLabel(label: RsLabel): RsLabelDecl? =
+        label.ancestors
+            .takeWhile { it !is RsLambdaExpr && it !is RsFunction }
+            .mapNotNull { it as? RsLabeledExpression }
+            .map { it.labelDecl }
+            .find { it.name ==  label.quoteIdentifier.text}
+
+    fun resolveLifetime(lifetimeRef: RsLifetime): RsLifetimeDecl? {
+        if (lifetimeRef.isPredefined) return null
+        return lifetimeRef.ancestors
+            .mapNotNull {
+                when (it) {
+                    is RsTypeAlias -> if (it.parent is RsImplItem || it.parent is RsTraitItem) null else it.typeParameterList
```

Comment:
Why do we need `if (it.parent is RsImplItem || it.parent is RsTraitItem)` condition? 

We can just resolve this, and annotate lifetimes in associated types as forbidden (or just wait until they implement that HKT RFC :) )